app/services/users.js

```diff
@@ -6,10 +6,15 @@ const bcrypt = require('bcrypt'),
   saltRounds = 10;
 
 exports.create = (user) => {
-  return bcrypt.hash(user.password, saltRounds).then((hash) => {
-    user.password = hash;
-    return orm.models.user.create(user).catch((err) => {
-      throw errors.defaultError(err.message);
+  return bcrypt.genSalt(saltRounds).then((authenticationCode) => {
+    user.authenticationCode = authenticationCode;
+    return bcrypt.hash(user.password, saltRounds).then((hash) => {
+      user.password = hash;
+      return orm.models.user.create(user).catch((err) => {
+        throw errors.defaultError(err.message);
+      });
+    }).catch((err) => {
```

Comment:
You can avoid this catch, the one below will handle all the error this one is catching.